src/app/[locale]/page.tsx

```diff
@@ -27,35 +28,66 @@ export default function Index() {
     lat: 0,
     lng: 0,
   });
+  const [searchCenter, setSetSearchCenter] = useState<{ lat: number; lng: number }>({
+    lat: 0,
+    lng: 0,
+  });
+  const [findme, setFindme] = useState<DeviceLocationType>(DeviceLocationType.SearchCenter);
+  const [dbUserSettings, setDbUserSettings] = useState<IUserSettings | null>(null);
   const [zoomLevel, setZoomLevel] = useState(2);
   const [locationError, setLocationError] = useState<string | null>(null);
   const [searchQuery, setSearchQuery] = useState<string>('');
   const [isSearchClicked, setSearchClicked] = useState(false);
   const [searchResults, setSearchResults] = useState<any[]>([]);
 
-  const { isSigningInUser } = useContext(AppContext);
+  const { isSigningInUser, currentUser, autoLoginUser } = useContext(AppContext);
 
   // Default map center (example: New York City)
   const defaultMapCenter = { lat: 20, lng: -74.006 };
 
   useEffect(() => {
-    const fetchLocationOnLoad = async () => {
+    if (!currentUser) {
+      logger.info("User not logged in; attempting auto-login..");
+      autoLoginUser();
+    }
+
+    const getUserSettingsData = async () => {
       try {
-        const location = await fetchUserLocation();
-        setMapCenter(location.origin);
-        setZoomLevel(location.radius);
-        logger.info('User location obtained successfully on initial load:', {
-          location,
-        });
+        const data = await fetchUserSettings();
+        if (data) {
+          console.log('Fetched user settings data successfully: ', data.findme)
+          logger.info('Fetched user settings data successfully:', { data });
+          setDbUserSettings(data);
+          setSetSearchCenter(data.search_map_center.coordinates)
+        } else {
+          logger.warn('User Settings not found.');
+          setDbUserSettings(null);
+        }
       } catch (error) {
-        logger.error('Error getting location on initial load.', { error });
-        setMapCenter(defaultMapCenter);
-        setZoomLevel(2);
+        logger.error('Error fetching user settings data:', { error });
       }
     };
+    getUserSettingsData();
+  }, [currentUser]);
+
+  // useEffect(() => {
```

Comment:
Can we do away with this commented out block? If possible do away with all the commented out codes

Reply:
Of course! 👌